src/btree/bt_handle.c

```diff
@@ -881,8 +882,20 @@ __btree_get_last_recno(WT_SESSION_IMPL *session)
         return (WT_NOTFOUND);
 
     page = next_walk->page;
-    btree->last_recno = page->type == WT_PAGE_COL_VAR ? __col_var_last_recno(next_walk) :
-                                                        __col_fix_last_recno(next_walk);
+    last_recno = page->type == WT_PAGE_COL_VAR ? __col_var_last_recno(next_walk) :
+                                                 __col_fix_last_recno(next_walk);
+
+    /*
+     * If we skipped reading the page into memory, it's possible that next_walk points to an empty
```

Comment:
This sentence is odd to me. If we skip reading the page, the page we see will be an empty page. I would rephrase it as "If the last page is deleted and it is globally visible, we skip reading the page from disk and instead instantiate an empty page in memory."

Reply:
Fair enough - fixed in 4d490b6.